### DISCUSSION TITLE
Option to pass the CSS rules in configuration

### DISCUSSION THREAD

**Discussion by Yonier** [niklasvh/html2canvas#2772] [Ideas]
It would be a good idea to pass the CSS rules of the HTML element so the library doesn't have to search in the whole DOM for the CSS rules of the element. Its a good idea when you have a heavy page with lots of CSS rules, like in my case the library gives a very good result but it takes a lot of time to generate the Canvas for a simple html block (a short text with font sizes)... so with this idea the library would work even quickly.

I hope this could be possible to implement, thank you!.

**Comment by niklasvh** [comment]
Could you expand a little on how you'd envision this to look like? Would you just provide the styles for the to a single element, or are you proposing that you could paste contents of a stylesheet that we'd embed within a `style` element?

**Reply by Yonier** [reply]
Yes, that you could pass the path of the stylesheet of the element, something like:

`html2canvas(element, { styleSource/cssSource: '@/assets/canvas.css' })`

**Comment by Yonier** [comment]
Another thing I think it could work for better improvement is that when creating the cloned iframe it would be good to only clone the passed element and not the whole DOM elements, because in my case is cloning almost 40k lines when i'm using less than 200 lines for the element I pass, so i think with this it would generate the canvas faster.
